test/Microsoft.ML.Benchmarks/FeaturizeTextBench.cs

```diff
@@ -0,0 +1,139 @@
+
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML.Data;
+using BenchmarkDotNet.Attributes;
+using Microsoft.ML.Transforms.Text;
+using Xunit;
+
+namespace Microsoft.ML.Benchmarks
+{
+    [Config(typeof(TrainConfig))]
+    public class FeaturizeTextBench
+    {
+        private MLContext mlContext;
+        private IDataView dataset;
+        private static int numColumns = 1000;
+        private static int numRows = 300;
+        private static int maxWordLength = 15;
+
+        [GlobalSetup]
+        public void SetupData()
+        {
+            Path.GetTempFileName();
+            mlContext = new MLContext(seed: 1);
+            var path = Path.GetTempFileName();
+            Console.WriteLine($"Created dataset in temporary file:\n{path}\n");
+            path = CreateRandomFile(path);
+
+            var columns = new List<TextLoader.Column>();
+            for(int i = 0; i < numColumns; i++)
+            {
+                columns.Add(new TextLoader.Column($"Column{i}", DataKind.String, i));
+            }
+
+            var textLoader = mlContext.Data.CreateTextLoader(new TextLoader.Options()
+            {
+                Columns = columns.ToArray(),
+                HasHeader = false,
+                Separators = new char[] { ',' }
+            });
+
+            dataset = textLoader.Load(path);
+        }
+
+        [Benchmark]
+        public ITransformer TrainFeaturizeText()
+        {
+            var textColumns = new List<string>();
+            for (int i = 0; i < 20; i++) // Only load first 20 columns
+            {
+                textColumns.Add($"Column{i}");
+            }
+
+            var featurizers = new List<TextFeaturizingEstimator>();
+            foreach (var textColumn in textColumns)
+            {
+                var featurizer = mlContext.Transforms.Text.FeaturizeText(textColumn, new TextFeaturizingEstimator.Options()
+                {
+                    CharFeatureExtractor = null,
+                    WordFeatureExtractor = new WordBagEstimator.Options()
+                    {
+                        NgramLength = 2,
+                        MaximumNgramsCount = new int[] { 200000 }
+                    }
+                });
+                featurizers.Add(featurizer);
+            }
+
+            IEstimator<ITransformer>  pipeline = featurizers.First();
+            foreach (var featurizer in featurizers.Skip(1))
+            {
+                pipeline = pipeline.Append(featurizer);
+            }
+
+            var model = pipeline.Fit(dataset);
+            var memoryUsage = GC.GetTotalMemory(true);
+            Console.WriteLine($"Memory Used: {memoryUsage/1000000:0,0.00}MB");
+            Assert.True(memoryUsage < 1000000000, $"This benchmark should use less than 1GB of memory, but it's using {memoryUsage/1000000:0,0.00}MB"); // Memory usage should be less than 1GB after PR #4576
```

Comment:
@harishsk this is the assertion you asked me to add to the benchmark. In my machine, without the changes made in this PR, the value of `memoryUsage` is over 2.5GB... with the changes of this PR it's less than 200MB. So I didn't know where to put the threshold, and I simply chose 1GB. Don't know if that's ok to you. Thanks.

Reply:
1GB maybe a bit high. How about 2x the memory usage you are seeing on your machine? If memory usage on a different machine exceeds two times what you are seeing on your machine, then it is reason to investigate what is happening.

Reply:
Ok, so I will change it into 400MB. Thanks!

Reply:
Sorry, when running the test on my machine memoryUsage is actually, at most 120MB, so I will fix the threshold into 240MB.

Reply:
@adamsitnik Is there a way to make assertions about the stabilized outcome of the benchmark?